tests/framework/test_engine.py

```diff
@@ -243,14 +257,105 @@ def test_SimulationContext_finalize(SimulationContext, base_config, components):
     assert listener.simulation_end_called
 
 
-def test_SimulationContext_report(SimulationContext, base_config, components):
-    # TODO [MIC-4994] Update with new results processing
+def test_SimulationContext_report(SimulationContext, base_config, components, tmpdir):
     sim = SimulationContext(base_config, components)
     sim.setup()
     sim.initialize_simulants()
     sim.run()
     sim.finalize()
-    metrics = sim.report()
-    assert metrics["test"] == len(
+    sim.report(tmpdir)
+
+    metrics = pd.read_csv(tmpdir / "test.csv")
+    assert len(metrics["value"].unique()) == 1
+    assert metrics["value"].iat[0] == len(
         [c for c in sim._component_manager._components if isinstance(c, MockComponentB)]
     )
+
+
+def test_SimulationContext_report_output_format(tmpdir):
+    """Test report output is as expected"""
+    results_root = Path(tmpdir)
+    components = [
+        Hogwarts(),
+        HousePointsObserver(),
+        QuidditchWinsObserver(),
+        HogwartsResultsStratifier(),
+    ]
+    finished_sim = run_simulation(components=components, configuration=CONFIG)
+    finished_sim.report(results_root)
+
+    # The observers are based on number of steps - extract how many steps were run
+    time_dict = finished_sim.configuration.time.to_dict()
+    end_date = _convert_to_datetime(time_dict["end"])
+    start_date = _convert_to_datetime(time_dict["start"])
+    num_steps_float = (end_date - start_date).days / time_dict["step_size"]
+    # round up
+    num_steps = int(num_steps_float) + 1 if num_steps_float % 1 else num_steps_float
```

Comment:
nit: the `else` makes this return a float  (e.g. 5.0) instead of 5. Is that what you want? Does math.ceil() not work?

Reply:
Eh, I was honestly just too lazy to go to the top to import a new library. I'll use `math.ceil` though; it's more readable